spring-cloud-zuul-ratelimit-core/src/main/java/com/marcosbarbero/cloud/autoconfigure/zuul/ratelimit/config/properties/RateLimitProperties.java

```diff
@@ -16,35 +16,62 @@
 
 package com.marcosbarbero.cloud.autoconfigure.zuul.ratelimit.config.properties;
 
-import static com.marcosbarbero.cloud.autoconfigure.zuul.ratelimit.config.properties.RateLimitProperties.PREFIX;
-
-import java.util.LinkedHashMap;
+import com.google.common.collect.Lists;
+import com.google.common.collect.Maps;
+import java.util.List;
 import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.TimeUnit;
+import javax.validation.constraints.NotNull;
 import lombok.Data;
+import lombok.NoArgsConstructor;
 import org.springframework.beans.factory.annotation.Value;
 import org.springframework.boot.context.properties.ConfigurationProperties;
+import org.springframework.validation.annotation.Validated;
 
 /**
  * @author Marcos Barbero
  */
 @Data
-@ConfigurationProperties(PREFIX)
+@Validated
+@NoArgsConstructor
+@ConfigurationProperties(RateLimitProperties.PREFIX)
 public class RateLimitProperties {
 
     public static final String PREFIX = "zuul.ratelimit";
 
-    private Map<String, Policy> policies = new LinkedHashMap<>();
-
+    private Policy defaultPolicy;
+    @NotNull
+    private Map<String, Policy> policies = Maps.newHashMap();
     private boolean behindProxy;
     private boolean enabled;
-
+    @NotNull
     @Value("${spring.application.name:rate-limit-application}")
     private String keyPrefix;
-
-    private Repository repository;
+    @NotNull
+    private Repository repository = Repository.IN_MEMORY;
 
     public enum Repository {
-        REDIS, CONSUL, JPA
+        REDIS, CONSUL, JPA, IN_MEMORY
     }
 
+    public Optional<Policy> getPolicy(String key) {
+        return Optional.ofNullable(policies.getOrDefault(key, defaultPolicy));
+    }
+
+    @Data
+    @NoArgsConstructor
+    public static class Policy {
+
+        @NotNull
+        private Long refreshInterval = TimeUnit.MINUTES.toSeconds(1L);
+        @NotNull
+        private Long limit;
+        @NotNull
+        private List<Type> type = Lists.newArrayList();
```

Comment:
Why not to use `Collections.emptyList()` instead?

Reply:
Spring properties is inserting to the map, not creating new one